src/widget.js

```diff
@@ -1,23 +1,140 @@
+
+const defaults = {
+  size: 'icon',
+  url: window.location.href,
+  inlineCss: false,
+  version: 1,
+  all: false
+};
+
+const templates = {
+  icon: `<div class="superhero-button-link-holder">
+            <a id="superhero-tip-link" class="superhero-button-link" target="_blank">
+              <img src="../img/v1-icon.png" />
+            </a>
+          </div>`,
+  small: `<div class="superhero-button-link-holder">
+            <a id="superhero-tip-link" class="superhero-button-link" target="_blank">
+              <img src="../img/v1-icon.png" />
+              <span>Donate</span>
+            </a>
+          </div>
+          <div class="superhero-button-tips-amount"><span id="tips">0</span><span class="ae">AE</span></div>`,
+  medium: `<div class="superhero-button-link-holder">
+            <a id="superhero-tip-link" class="superhero-button-link" target="_blank">
+              <img src="../img/v1-icon.png" />
+              <span>Donate Now</span>
+            </a>
+          </div>
+          <div class="superhero-button-tips-amount"><span id="tips">0</span><span class="ae">AE</span></div>`,
+  large: `<div class="superhero-button-tips-amount"><span id="tips">0</span><span class="ae">AE</span></div>
+          <div class="superhero-button-link-holder">
+            <a id="superhero-tip-link" class="superhero-button-link" target="_blank">
+              <img src="../img/v1-icon.png" />
+              <span>Donate Now</span>
+            </a>
+          </div>`
+};
+
+class SuperheroButton {
+  
+  constructor(element, options) {
+    const opt = { ...defaults, ...options };
+    this.insertStyles(opt);
+    const el = this.isDomNode(element) || this.isNodeList(element) ? element : opt.all ? document.querySelectorAll(element) : document.querySelector(element);
+    if (this.isNodeList(el)) {
+      let buttonInstances = [];
+      el.forEach(e => {
+        const instance = this.createButtonInstance(opt);
+        buttonInstances.push(instance);
+        e.replaceWith(instance);
+      });
+      this.instance = buttonInstances;
+    } else {
+      this.instance = this.createButtonInstance(opt);
+      el.replaceWith(this.instance);
+    }
+  }
+
+  isDomNode(element) {
+    return element instanceof Element || element instanceof HTMLDocument
   }
 
-  for(var f in iframes) {
-    document.querySelectorAll("a.superhero-tip-button")[0].replaceWith(iframes[f]);
+  isNodeList(nodes) {
+    return NodeList.prototype.isPrototypeOf(nodes);
   }
-});
\ No newline at end of file
+
+  getInstance() {
+    return this.instance;
+  }
+
+  createButtonInstance({ size, url, account }) {
+    if (!templates[size]) throw new Error('Unsupported size');
+    var button = document.createElement('button');
+    button.innerHTML = templates[size];
+    button.className = `superhero-button-container superhero-button-${size}`;
+    const isMobile = navigator.userAgent.match(/(iPad)|(iPhone)|(iPod)|(android)|(webOS)/i);
```

Comment:
In Base aepp we were detecting a mobile device just by:
```suggestion
    const isMobile = navigator.userAgent.includes('Mobi');
```
https://developer.mozilla.org/en-US/docs/Web/HTTP/Browser_detection_using_the_user_agent#Mobile_Tablet_or_Desktop